mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -429,10 +539,30 @@ struct ConvertVectorMaskedLoad final
     // Setting the part that originally was not effectively loaded from memory
     // to pass through.
     auto bitCast =
-        rewriter.create<vector::BitCastOp>(loc, op.getType(), newLoad);
-    auto select = rewriter.create<arith::SelectOp>(loc, op.getMask(), bitCast,
-                                                   op.getPassThru());
-    rewriter.replaceOp(op, select->getResult(0));
+        rewriter.create<vector::BitCastOp>(loc, newBitcastType, newLoad);
+
+    Value mask = op.getMask();
+    if (isUnalignedEmulation) {
+      auto newSelectMaskType =
+          VectorType::get(numElements * scale, rewriter.getI1Type());
+      // TODO: can fold if op's mask is constant
+      auto emptyVector = rewriter.create<arith::ConstantOp>(
+          loc, newSelectMaskType, rewriter.getZeroAttr(newSelectMaskType));
+      mask = insertSubvectorInto(rewriter, loc, op.getMask(), emptyVector,
+                                 *foldedFrontPaddingSize);
+    }
+
+    auto select =
+        rewriter.create<arith::SelectOp>(loc, mask, bitCast, passthru);
+
+    if (isUnalignedEmulation) {
+      auto extract =
+          extractSubvectorFrom(rewriter, loc, op.getType(), select,
+                               *foldedFrontPaddingSize, origElements);
+      rewriter.replaceOp(op, extract);
+    } else {
+      rewriter.replaceOp(op, select->getResult(0));
+    }
```

Comment:
optional nit: I think we can declare `Value result` and replace the op at the end, which is consistent with other patterns in this file. Also, it looks cleaner to me. E.g.,

```cpp
Value result = rewriter.create<arith::SelectOp>...
if (isUnaligned) {
  result = ...
}
rewriter.replaceOp(op, result);
```